src/query/query_parser/query_parser.rs

```diff
@@ -68,6 +68,9 @@ pub enum QueryParserError {
     /// The format for the date field is not RFC 3339 compliant.
     #[error("The date field has an invalid format")]
     DateFormatError(chrono::ParseError),
+    /// The format for the facet field is invalid.
+    #[error("The facet field is malformed")]
```

Comment:
I'm not sure, but I think this error should forward the original error message or the user would not see which part of the query failed

Reply:
like this `#[error("{0}")]`